h2o-automl/src/main/java/ai/h2o/automl/modeling/XGBoostSteps.java

```diff
@@ -215,8 +216,51 @@ public Map<String, Object[]> prepareSearchParameters() {
 
     }
 
+    static class XGBoostGBLinearGridStep extends XGBoostGridStep {
+        
+        public XGBoostGBLinearGridStep(String id, AutoML autoML) {
+            super(id, autoML, false);
+        }
+
+
+        @Override
+        public XGBoostParameters prepareModelParameters() {
+            return XGBoostSteps.prepareModelParameters(aml(), false);
+        }
+
+        @Override
+        public Map<String, Object[]> prepareSearchParameters() {
+            Map<String, Object[]> searchParams = new HashMap<>();
+
+            /* 
+            // not supported/exposed in our xgboost yet
+            if (aml().getBuildSpec().build_control.isReproducible()) {
+                searchParams.put("_updater", new String[] {"coord_descent"});
+                searchParams.put("_feature_selector", new String[] {"cyclic", "greedy"});  // TODO: check if others are deterministic
+                            } else {
+                searchParams.put("_updater", new String[] {"shotgun", "coord_descent"});
+                searchParams.put("_feature_selector", new String[] {"cyclic", "shuffle", "random", "greedy", "thrifty"});
+            }
+            int ncols = aml().getTrainingFrame().numCols() - (aml().getBuildSpec().getNonPredictors().length +
+                    (aml().getBuildSpec().input_spec.ignored_columns != null ? aml().getBuildSpec().input_spec.ignored_columns.length : 0));
+
+            searchParams.put("_top_k", IntStream.range(0, ncols-1).boxed().toArray(Integer[]::new));
```

Comment:
Nice to add this in now.  You can uncomment it once Adam exposed those parameters.